### PR TITLE
Improved sidebar links for navigability

### PR DESCRIPTION
**Pull request by wedebe**
Proper sidebar links

This change implements url improvements for sidebar nav links
to allow for back/fwd page navigation and bookmarking

### CODE REVIEW

File: sourcefiles/js/openwebif.js [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ function editTimer(serviceref, begin, end) {
 							if (typeof timer.allow_duplicate !== 'undefined')
 							{
 								$('#allow_duplicate').prop("checked", timer.allow_duplicate==1);
-								autoadjust: ($('#autoadjust').is(':checked')?"1":"0"),
+								//autoadjust: ($('#autoadjust').is(':checked')?"1":"0"),
```
Comment (jbleyel) [inline sourcefiles/js/openwebif.js:1253]: This breaks the VTi functionality.

File: sourcefiles/js/openwebif.js [hunk 2]
```diff
@@ -1529,7 +1529,6 @@ function InitTVRadio(epgmode)
 
 	if (tv===true) {
 		var parts=window.location.href.toLowerCase().split("#");
-		window.location.hash="";
 		if (parts[1] == 'tv') {
```
Comment (jbleyel) [inline sourcefiles/js/openwebif.js:1532]: i think this is necessary

File: sourcefiles/js/openwebif.js [hunk 3]
```diff
@@ -1611,7 +1610,6 @@ function InitBouquets(tv)
 
 	if (tv===true) {
 		var parts=window.location.href.toLowerCase().split("#");
-		window.location.hash="";
```
Comment (jbleyel) [inline sourcefiles/js/openwebif.js:1614]: i think this is necessary

File: sourcefiles/js/openwebif.js [hunk 4]
```diff
@@ -1731,12 +1728,14 @@ function directlink()
...
 		case 'epgr':
+		case 'myepg':
+		case 'timers':
+		case 'satfinder':
+		case 'boxinfo':
+		case 'webtv':
 			lnk='ajax/' + p;
 			break;
 	}
-	if(p != 'tv') {
```
Comment (jbleyel) [inline sourcefiles/js/openwebif.js:1737]: i think this is necessary

**Review by jbleyel — CHANGES_REQUESTED** [review]

**Comment by jbleyel** [comment]
This change also affects the non responsive UI.
Do you have test this?

**Comment by wedebe** [comment]
> This change also affects the non responsive UI.
> Do you have test this?

I've tested the sidenav links in the non-responsive ui and they continue to work as they don't use hash-based hrefs.

**Comment by jbleyel** [comment]
> > This change also affects the non responsive UI.
> > Do you have test this?
> 
> I've tested the sidenav links in the non-responsive ui and they continue to work as they don't use hash-based hrefs.

This is not 100% correct.
I have implement the direct link feature to open the page with a specific target page.
The removal of the hash after loading is needed as i remember. -> `window.location.hash="";`

**Comment by jbleyel** [comment]
Please check this -> 
https://github.com/E2OpenPlugins/e2openplugin-OpenWebif/commit/483931df18d752b5e089b2e9b49f74a00a54e05a

**Comment by jbleyel** [comment]
And this.

https://github.com/E2OpenPlugins/e2openplugin-OpenWebif/commit/821e2dd67546a1a9359cf7122680663cc2c3ccea#diff-5349448c9de06a950c31cdc16b9528af

It's aprox 4 years ago. I don't 100% remember.

**Comment by jbleyel** [comment]
I'm not sure but i think this `window.location.hash="";` was a design thing.
I don't want #xxx in the link field of the browser.
And some links do not work if you do not clear the hash.

**Comment by wedebe** [comment]
> I'm not sure but i think this `window.location.hash="";` was a design thing.
> I don't want #xxx in the link field of the browser.
> And some links do not work if you do not clear the hash.

Let me know (if possible) which links this applies to and I'll investigate. I did check out the directlink function, perhaps I didn't call it in the right circumstances but again I'm happy to check it out.

**Pull request merged**